### ISSUE TITLE
Add Botania

### ISSUE THREAD

**Issue by combusterf** [GTNewHorizons/GT-New-Horizons-Modpack#9314]
### Your GTNH Discord Username

Combuster

### Your Pack Version

2.0.x.x-2.1.x.x

### Mod Name

Botania

### The Mod to Add

Botania is a tech mod by Vazkii. It includes many mechanics not currently in the pack and has been requested for addition on previous occasions.

### Your Reason

I was invited to do this :wink: 

### Preliminary Investigation

I have forked Botania and made some minor technical improvements for compatibility. The bulk of this relates to gregification, where recipes are moved into GT machines, and the ingame help/lexicon being dependent on many such recipes being vanilla crafting ones. The lexicon thus remains functional and lists the correct (modded) recipes when compatible, or a stub page if the recipe has been gregified or removed. The features otherwise match that of the latest public 1.7.10 release.

Relevant links:
- [Github repository](https://github.com/combusterf/Botania-expert)
- [Prebuilt jar](https://fuzzycraft.net/downloads/Botania-r1.8-250.jar)
- [Prebuilt deobfuscated jar](https://fuzzycraft.net/downloads/Botania-r1.8-250-deobf.jar) (developers only)

The second part is a companion mod that performs the gregification and handles the bindings between Botania and Gregtech/Thaumcraft, with some excursions to Natura, Witchery, Twilight Forest, Draconic evolution and the GTNH core mod. The current version already provides gregified versions of recipes and accounts for the level-tiering.

Relevant links: 
- [Github repository](https://github.com/combusterf/Botanic-horizons)
- [Prebuilt jar](https://fuzzycraft.net/downloads/BotanicHorizons-0.9.0.jar)

Both mods together can be added to a stock GTNH modpack and provide a fully functional and gregified addition to the pack with no other errors observed. The result is backwards compatible: I have been playing on the public servers for the past months with these two mods installed. So barring any oversight - testing your own code is the worst QA - we are ready to go.

-----

Progression overview:
1. Flower spawns have been disabled to prevent easy access. Mushrooms spawn in the overworld and require the user to go spelunking for them. Mushrooms can be used as substitutes for flowers, and this allows the user to start Botania in the stone age - if they dare.
2. Early mana infusions have been nerfed. The player first has to make the diluted pool. This pool has limited capacity and can't make mana pearls. Manasteel is made from GT steel rather than iron, and is thus gated by the BBF.
3. Magical flowers need a worldgen flower as an ingredient. Depending on recipe this is either by flower colour, or a specific genus. There are a few methods for renewable flower production.
4. Diamonds convert into mana diamonds at a rate comparable to their quality. Only exquisite diamonds can be converted in the diluted pool. Getting these requires the LV sifter, and is thus gated by LV technology.
5. Thaumcraft in MV brings the first convenience set. The regular pool can be created which unlocks the remaining mana infusions. Thaumcraft recipes allow conversion from mushrooms into flowers, after which they can be multiplied. Missing colours still have to be found through spelunking.
6. The alchemical catalyst is available as a research and gives access to prismarine
7. The player can decide to venture into Witchery to get floral fertiliser early and get the remaining colours from relative safety. Thaumcraft infusion can be done at HV without fertiliser, and draconium dust is available later as the designated "easy" feedstock for floral fertiliser. The jaded amaranthus can be obtained between HV and draconium if the player has the necessary colours
8. The runic altar is crafted from the thaumcraft infusion matrix. T1 and T2 runes require HV resources, 
9. T3 runes require titanium and are thus gated to EV gate
10. Once the player has all runes, terrasteel becomes available in EV.
11. The elven gateway core is gated by IV-tier materials, as well as some Twilight Forest resources. 
12. The gaia boss is also in IV, but this is forbidden technology and requires the Eldritch tab. The player is free to wait until late-IV/early-LuV tech to do the tear quest
13. Some OP items have tiers past IV. The Shard of Laputa has been nerfed big-time, and is the designated Botania item for the endgame goals tab.
14. The everlasting guilty pool has a recipe for the lucky few with infinity ingots.

-----

Batteries not included:
- I have not written a questline for the quest book. 
- Existing worlds compatibility doesn't exist. The easiest way to bootstrap this is by selling white mushrooms and nightshades in the coins tab. There is currently no retrogen.
- Some things that would better be done in other mods: GT knows *some* things about manasteel/terrasteel/elementium and is probably the best place to get plates and screws etc. After that the recipes involving these can be updated to look more GT:NH style.

-----

Sneak preview:
```
So the rumours about the elves were true. Well, *were* true. They're still mostly true. and what isn't true used to be true.

Turns out they have since been talking to a person going by the name of Olafe, which has since sparked a industrial revolution in the Elven lands. That sadly means that they now insist on receiving the latest fashion in machinery to fuel their ever-expanding factories, in exchange for the elven materials that are now rapidly going out of fashion.

Someone needs to have a word with this guy...
```
![image](https://user-images.githubusercontent.com/8459776/147837056-5236dc8d-4f2a-4266-af61-e7a697e4376e.png)

-----

I wish you all a happy new year!

### Final Checklist

- [X] This mod is released under an open source license. Yes, this is required now.
- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this mod, please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [X] I believe there is nothing similar in the pack already, or the existing solution isn't good enough.
- [X] I understand this change request may not attract enough attention and thus not be implemented.
- [X] I understand this change request may be rejected due to other community members thinking it's inappropriate.
- [X] I believe this feature would make the pack better.
- [X] I'm the official maintainer or original author of this mod for minecraft 1.7.10.

**Comment by 0lafe** [comment]
This seems really cool! One question I have is related to
 >where recipes are moved into GT machines

Does this mean you wouldn't be using the petal crafting pool, or runic altar? Personally I think the more unique mechanics of botania crafting is a lot of what makes it unique.

**Comment by combusterf** [comment]
That's referring to things like blocks of manasteel - which should be a compression recipes in line with all the others, rather than a 3x3 craft. Removing the original recipe also removes it from the lexicon.

**Comment by bombcar** [comment]
> Existing worlds compatibility doesn't exist.

does this include new chunks created after the mod is added

**Comment by combusterf** [comment]
> does this include new chunks created after the mod is added

Newly created chunks will inclued the expected worldgen. This doesn't help on the officials or other servers that have the entire map pregenerated.

**Comment by yukieiji** [comment]
Previous discussion: #6446, #731

**Comment by bombcar** [comment]
Perhaps certain things that are locked behind thaumcraft could also alternatively be locked behind something of similar difficulty in botania

**Comment by Regian24** [comment]
why

**Comment by Florexiz** [comment]
Tbh I can't recall anything useful from botania that doesn't already exist in pack
So yeah, why

**Comment by FlandreS** [comment]
Mana enchanting is decently cool.
Teru Teru bozu offers some weather control
Incense could be used for warp ward or as an accessory to blood magic.
The wands could have decent uses in the early-mid game with the rod of the skies being the glider's best friend.
Horn of the canopy/wild are good for clearing fields and leaves.
Sashes will let the speed demons continue their quest for unplayable levels of SPEED.
Bauble that offers a signifigant reach buff.

I'm a Botania novice/apprentice at best but there's a lot of unique uses, there's more than I can think of off the top of my head for sure. The only concerns I have are that some items may be tiered too late to be of real use. Haven't taken a look at recipes yet but if wands take runes and are post-HV then I could see them being unused.

Even if Botania's base items are something individuals consider uninteresting, it has a unique power and crafting system that could be used to craft items outside of itself if there was interest in tying some recipes to Botania progression.

**Comment by combusterf** [comment]
There are few things that it achieves that aren't covered by other mods. In comparison, there are a lot of things that do this much more *conveniently* compared to other mods.  http://files.combuster.nl/botania.html

As for not adding, we're past that: https://discord.com/channels/181078474394566657/465207956745486336/739521806490402847 
![image](https://user-images.githubusercontent.com/8459776/148119148-d7af9770-d79f-4882-b852-3193596f086e.png)

**Comment by bombcar** [comment]
It's clear to me that @combusterf has taken the time to work out where this may slot in, and I think we should attempt it for 2.1.3.0 or whatever the next big release is.

**Comment by Minepolz320** [comment]
Good!

**Comment by Barhandar** [comment]
"Conveniently" is complete opposite to the spirit of the pack (which often is "be as obnoxious as possible"). Botania is essentially severely casualized knockoff Blood Magic*, with very few things that BM doesn't have (but can be potentially added, like Warp Ward potions), the chief difference is that it uses bauble slots instead of inventory slots; plus a number of features from Vazkii's previous mod, that being Thaumic Tinkerer. Popular vote for its inclusion was a bad idea, as is inclusion itself.

Also, I've read the "full documentation" and I have to say that you're _not qualified_ for balancing Botania or adding it to the pack. Overgrowth Seed, or rather the soil it makes, doubles the generating flower output, a lot of things have options in the pack that you don't know about like the Slimalyzer as alternative to Slime in a Bottle, you've put Fruit of Grisaia one (two) tiers above Biscuit of Totality even though the primary difference is fruit takes half as much mana and doesn't waste it, et cetera.

*blaster with flare lens: sigil of the blood lamp; rod of bifrost: sigil of the phantom bridge; ring of the mantle: sigil of the fast miner; ring of odin: sigil of elemental affinity; extrapolated bucket: void sigil; bubbell: ritual dome of suppression; agricarnation: ritual/sigil of the green grove; horn of the wild: harvest goddess sigil, et cetera

**Comment by Barhandar** [comment]
P.S. Github refuses to properly work so I can't edit the message, but
>Some OP items have tiers past IV. The Shard of Laputa has been nerfed big-time, and is the designated Botania item for the endgame goals tab.

are... are you aware what it _does_?

**Comment by YannickMG** [comment]
Would you mind being a bit more constructive and less dismissive in your feedback? Your feedback could be very helpful if it tried to be.

**Comment by YannickMG** [comment]
If you have specific balance recommendations to make or if say, for instance, some items are too OP to be balanced and should be removed, a new ticket would be the best way to let us know.

**Comment by Chaoschaot232** [comment]
Talked with @Dream-Master and doing the Botania Implementation for this by using the existing Botanic Horizons Mod so I'll share a few things here to talk about and to checkout when they are written within the mod. I hope that you get the imagination to this adn give it a chance:

Botania will get sliced down with an own Questline and it's Book will be splitted so that you must collect the knowledge on how to use what item. Without this knowledge => No Usage! You need to progress through the BT-Questline to get Key-Items which you must then trade in the Elven Portal.
Q: How should I trade when I don't have the Portal now?
A: Until that point, you'll get the "Knowledge Pages" as Quest Reward. Harder Stuff is gated behind this in addition then.

Cosmetic Blocks will go into Chisel instead of using the Crafting Grid. They'll also get there own little Quest within the Book.
As they are only for Cosmetic purpose there is no need in gating them behind anything. Only the Dreamwood will be gated
because it is also an ingredient for better stuff like the Elven/Gaia Mana Spreader.

For the Mana/Terra/Elementium Ingots ... tbh. but NO DIRECLTY EBFing them as I refere them to be "Magical Ingots" which means that I am going to let you trade them in over the Elven Portal but the price is high enough:

Manasteel (Botania - Gate: EBF/MV/Stainless Steel):

Ratio: 1:3 (or if I would be a badass 1:1 if you all prefere but maybe this is a bit to hard nerfed ^^)
Input: Block of Stainless Steel
Output: Manasteel Ingot (YES, only the Ingot!)

TerraSteel (Botania):

Made over the original Methode but with a much higher amount of Mana beeing used and at last GT Materials. We should after all not go and break intended mod specific mechanics only because of having GT/Thaumcraft in the first place so not all must be directly done im MBs and/or Machines from GT tbh.

Elementium Ingot (Botania):

Ever noticed Avaritia? If not, you gonna hate me but you'll need it to get this valuable Item as I let you create different Botania exclusive Avaritia Cores which you are going to need for an Infusion ontop of this Step. Well yes, you are going to infuse a Terrasteel Ingot for this.

Armor + Armor: All are going into Avaritia Extreme Crafting!

Brews: Will be untuched for now

Blood Pendant + Incense Sticks: Infusion or did you assumed anything else?

Runes: Definitly harder to make and higher usage of Mana
Flowers: Usage of more Runes and GT Items which hopefully suite the Flower and also there power in terms of mana generated.

Lenses: Arcane Crafting - If you finde a few lenses to OP, tell me and they'll get nerfed to an Infusion or to an Avaritia Extreme Crafting Recipe (we have to less of them xP)

Shard of Laputa: I am twisted as it can be seen to help out whith building nice places but on the other side it simple makes floating islands so that you could mine ores faster an maybe safe. When this also works on plantes then ...
So either it will be an Infusion or an Avaritia ones. Upgrades can be Infusion or also Avaritia ones but I would like to add the same methode like the Terra Shatter has. So charging it with mana, making it not despawnable. If this will happen, be prepared fo a hughe amount of mana beeing consumed for this process.

Flügel Tiara: Creative Flight at the coast of mana? Truely smth. for the Endgame, Infusion and Avaritia Extreme Crafting.

Cooporea System: Not realy smth. what should be gated far behind anything. Could be a nice first replace of all these Drawers.

Flowers will not spawn, instead I fall back to the existing methode of turning mushrooms into flowers but with a slightly different usage of aspects. Why? Because you would need only one Flower for the Petal which you later can burrie on a gras block where they are then grow. Setting then up a Flower Farm shouldn't be any problem.

Hardest part would be the Balance for Gaja I and Gaja II where I would maybe need some help as it seems to be too easy to kill both of them.

Mana System:

I'll go and splitt between passive and active mana production but I'll hide a little secret on this which you have to find out on your own with the given hints in the quest line. For now I'll go out and rais up the decay value and lower down the mana generated from these Flower and regarding to there recipes I don't think that there will be large are wide usages.
The active ones will give the normal output but here with that mentioned secred which is ... A SECRET (for now).

There will be some items exclusivly created for this to work as I am also going to split the Mana and Gaia Pylon because they are strictly two items. A much better and powerful version of the Natura Pylon and they are "held in balance" with a ring and this ring will be created as ingredient for the recipes.

So this is for the theoretical side and yes, the newer versions of Botania already using a progression based system so I think and hope that this change should suite and fit in nicely.

I am open for any constructive argument to any recipe added which I'll share later on (not all, just a few) and also for adjustments to any other things but for now only this shared idea from my side. I please you to give me that chance and hopefully make Botania a great experience in this modpack :)

**Comment by combusterf** [comment]
DreamMaster has a lot of ideas, but not always a sense of balance or what the playerbase at large goes through.

> you must collect the knowledge on how to use what item.
Botania doesn't come with any architecture to gate crafts based on research. If you wish to implement this, you get the functional equivalent of a Thaumcraft add-on (and you might as well write it as a thaumcraft add-on at that point instead of keeping up appearances) which runs mostly afoul of the community opinion that they want less integration and less thaumcraft and its style. The lexicon is -with all the fixes I had to make already- still architecturally a mess.

> Cosmetic Blocks will go into Chisel instead of using the Crafting Grid
Barring the recipe conflicts that have already been fixed, adding chisel is not really a functional change and could just be a feature request on its own. Removing regular crafting is a bad idea because that breaks automation. Chisel's automation is a farce where you have to keep replacing chisels.

> Manasteel (Botania - Gate: EBF/MV/Stainless Steel):
There have been a lot of comments that various botania items should be earlier to be competitive. Stainless steel is only barely MV - it is one of the last gates to HV. By changing from steel to stainless, you bump the current tiering by not just one, but 2,5 tiers, and severely limit your balancing options. If Dream is out to have nerfs on top of all the nerfs that I've already done, use blocks of regular steel. 

> Terrasteel [...] Made over the original Methode but with a much higher amount of Mana beeing used and at last GT Materials
As posted in the original ticket, everything has been nerfed already. This is already done in the current version. Did you instead wanted to suggest to change the recipe itself to include more moving parts, such as titanium ingots?

> Ever noticed Avaritia? If not, you gonna hate me but you'll need it to get this valuable Item as I let you create different Botania exclusive Avaritia Cores which you are going to need for an Infusion ontop of this Step. 
Elementium equipment is already very late to the party. It comes with some fancy enchantments, but otherwise its just a nano set unlocking two tiers up. As a crafting ingredient you could just suggest something better to use with the alfheim portal instead of inflating the price of something comparatively worthless.

> I'll hide a little secret on this 
This could also be read as forcing stuff upon others without giving them a chance to audit it. 

So summarised:
- Dream's plans are not always in line with what the community wants. Almost none of this has been discussed previously
- Botanic Horizons is already nerfed significantly over vanilla Botania. Don't just slap another zero after every mana cost, that will just cause time/lag/cpu power gates nobody is interested in.
- Dream doesn't seem to play, and you haven't progressed past the earliest levels of GTNH according to your Discord stats. It's easy to become overexcited and forget that difficulty isn't just about nerfing stuff, specifically that things don't get nerfed to the point where the addition of Botania itself becomes laughable as a whole.

**Comment by Chaoschaot232** [comment]
> DreamMaster has a lot of ideas, but not always a sense of balance or what the playerbase at large goes through.

I must correct things here if they where understanded wrongly ... I have told with Dream over it and he stated to write it down here for discussing it. This doesn't means that it is going to be exactly that way ;)

> Botania doesn't come with any architecture to gate crafts based on research. If you wish to implement this, you get the functional equivalent of a Thaumcraft add-on (and you might as well write it as a thaumcraft add-on at that point instead of keeping up appearances) which runs mostly afoul of the community opinion that they want less integration and less thaumcraft and its style. The lexicon is -with all the fixes I had to make already- still architecturally a mess.

Did you have looked into later Version of Botania? Only making it progression based doesn't makes it comparable with Thaumcraft progression. You are getting "new Entries" when you do specific Botania related things so simply look this up.
Means, that it rivels next informations to you, when you have done smth. - or in my case, added a "Knowledgepage".
You know about Thornpages from BloodMagic? It is like that way, that my intension here ;)

> Barring the recipe conflicts that have already been fixed, adding chisel is not really a functional change and could just be a feature request on its own. Removing regular crafting is a bad idea because that breaks automation. Chisel's automation is a farce where you have to keep replacing chisels.

So? When was this? Main problem here then is that there was no client release newer than the current QF Pack and this is what I am currently using, tbh.

> There have been a lot of comments that various botania items should be earlier to be competitive. Stainless steel is only barely MV - it is one of the last gates to HV. By changing from steel to stainless, you bump the current tiering by not just one, but 2,5 tiers, and severely limit your balancing options. If Dream is out to have nerfs on top of all the nerfs that I've already done, use blocks of regular steel.

Well, this was also my idea so blame me for this ^^
For me it is fine to change it to a block of steel and like stated above I have only the QF pack, nothing less and nothing more.
When there where made so many things it would have been better to maybe release some minor packupdates otherwise I can't see already made changes.

> As posted in the original ticket, everything has been nerfed already. This is already done in the current version. Did you instead wanted to suggest to change the recipe itself to include more moving parts, such as titanium ingots?

Titanium ingots? So instead of using the already transformed Mana Steel Ingots into Terra Steel Ingots you are going to use Titanium Ingots here and the Aglomeration Plate? Then I would have no problems with this but hopefully it isn't done in the EBF

> Elementium equipment is already very late to the party. It comes with some fancy enchantments, but otherwise its just a nano set unlocking two tiers up. As a crafting ingredient you could just suggest something better to use with the alfheim portal instead of inflating the price of something comparatively worthless.

🤔 Using the Alfheim Portal sounds interesting. But then may I ask if there should be items thrown in the portal to get the Elementium things out or only minor ones which are then be used in a crafting recipe to use them there as additional step?

> This could also be read as forcing stuff upon others without giving them a chance to audit it.

Ok, you got the point. - My idea was to produce Mana activly based on the Biome. We would need to think over which Biome would suite the 100% + Bonus and which one would be 100% - Bonus. I have had this idea while standing near my RC Water Tanks which are depending on the humidty level to fill up faster or slower incl. some additional bonus when it is raining.

> Dream doesn't seem to play, and you haven't progressed past the earliest levels of GTNH according to your Discord stats. It's easy to become overexcited and forget that difficulty isn't just about nerfing stuff, specifically that things don't get nerfed to the point where the addition of Botania itself becomes laughable as a whole.

Can't tell how far/deep Dream has progressed but yes, I am in the ealiest levels of GTNH but mainly because I wait for the new changes like the Kevlar line. Don't want to rebuild too much stuff here in my base and maybe this is understandable.

Yes, I am a bit overexcited but I generally think too complicated (applies to iRL things too). But on the other side this is the place to discuss over any things even if they are trivial or a nice idea which could be added ;)
I am still learning to make thinks not complicated as they already are but I am failing often at this point 😞 

Finaly I am glad to read about your thoughts and concearns to finde at the end a base and a point where it can be implemented, tested and published :)

**Comment by Runakai1** [comment]
We did it folks. Botania is in!
